front/lib/api/assistant/jit_actions.ts

```diff
@@ -340,6 +334,50 @@ export async function renderConversationForModelJIT({
     }
   }
 
+  if (selected.length > 0) {
+    const { filesAsXML, hasFiles } = await listConversationFiles({
+      conversation,
+    });
+    if (hasFiles) {
+      const randomCallId =
+        "list_conversation_files_" + Math.random().toString(36).substring(7);
+      const functionName = "list_conversation_files";
+
+      const simulatedAgentMessages = [
+        // 1. We add a message from the agent, asking to use the files listing function
+        {
+          role: "assistant",
+          content: "Please list the files available to this conversation.",
+          function_calls: [
+            {
+              id: randomCallId,
```

Comment:
Did you try this on all major model providers ? I believe some of them enforce a specific format for fcall IDs.

Reply:
I tried gpt, mistral, claude.
Do you have some pointers where to look for potential issues ?

Reply:
maybe worth trying gemini as well but from what I remembered it was mistral and anthropic that had strict ID formats. Maybe they lifted this 👍